src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java9/InputAdvanceJava9TryWithResources.java

```diff
@@ -0,0 +1,87 @@
+//non-compiled with javac: Compilable with Java9
+package com.puppycrawl.tools.checkstyle.grammar.java9;
+
+/**
+ * Input for Advance Java 9 try-with-resources.
+ */
+public class InputAdvanceJava9TryWithResources implements AutoCloseable
+{
+    //Constructor
+    public InputAdvanceJava9TryWithResources(boolean throwException) {
+        if (throwException)
+            throw new RuntimeException("Initialization exception");
+    }
+
+    //Constructor
+    public InputAdvanceJava9TryWithResources() {
+        this(false);
+    }
+
+    //Main method
+    public static void main(String[] args) throws Exception {
+        InputAdvanceJava9TryWithResources v = new InputAdvanceJava9TryWithResources();
+
+        try (v.finalWrapper.finalField) {
+        }
+
+        try (new InputAdvanceJava9TryWithResources() { }.finalWrapper.finalField) {
+        }
+
+        try ((args.length > 0 ? v
+                : new InputAdvanceJava9TryWithResources()).finalWrapper.finalField) {
+        }
+
+        //More than one resource
+        InputAdvanceJava9TryWithResources i1 = new InputAdvanceJava9TryWithResources();
+        try (i1; InputAdvanceJava9TryWithResources i2 =
+                    new InputAdvanceJava9TryWithResources(true)) {
+        }
+
+        InputAdvanceJava9TryWithResources m1 = new InputAdvanceJava9TryWithResources();
+        try (m1; InputAdvanceJava9TryWithResources m2 = m1;
+                InputAdvanceJava9TryWithResources m3 = m2;) {
+        }
+
+        // Nested try
+        try {
+        } catch (Exception ex) {
+            try (v) {
+            }
+        }
+        // null test cases
+        InputAdvanceJava9TryWithResources n = null;
+        try (n) {
+        }
+
+        // anonymous class implementing AutoCloseable as variable in try
+        AutoCloseable a = new AutoCloseable() {
+            public void close() { };
+        };
+        try (a) {
```

Comment:
please explain how this code is related to issue we try to resolve.
how it is different from:
```
      InputAdvanceJava9TryWithResources n = null;
        try (n) {
```
parser do not care about values and logic in code.

If you just extending our code base with more Input for better coverage - awesome, but please move them to separate Input file. Here we need to focus on all possible cases of field usage in `try`

Reply:
I still do not see separate input file.

Reply:
As you said there is no need of this code, so i have removed the line and did not create a sperate file. @romani